python/lsst/cell_coadds/_grid_container.py

```diff
@@ -0,0 +1,201 @@
+
+        Parameters
+        ----------
+        index : `~lsst.skymap.Index2D`
+            Index of the cell to return.
+        default : T
+            Value to return if the cell is not set.
+
+        Returns
+        -------
+        value : T
+            The value at the given index, or the default value if not set.
+        """
+        return self._cells.get(index, default)
+
+    @property
+    def shape(self) -> Index2D:
+        """Number of cells in the container in each dimension."""
+        return self._shape
+
+    @property
+    def offset(self) -> Index2D:
+        """Index of the first cell in the container."""
+        return self._offset
+
+    @property
+    def size(self) -> int:
+        """Return the number of cells expected in the container.
+        This does not indicated the number of cells that have been filled.
+        Use len() instead.
+        """
+        return self._shape.x * self._shape.y
+
+    @property
+    def first(self) -> T:
+        """The cell at the lower left corner of the container."""
+        return self._cells[self.offset]
+
+    @property
+    def last(self) -> T:
+        """The cell at the upper right corner of the container."""
+        return self._cells[Index2D(x=self.offset.x + self.shape.x - 1, y=self.offset.y + self.shape.y - 1)]
+
+    def subset_overlapping(self, grid: UniformGrid, bbox: geom.Box2I) -> GridContainer:
+        """Return a new GridContainer with cells that overlap a bounding box.
+
+        Parameters
+        ----------
+        grid : `~lsst.cell_coadds.UniformGrid`
+            Grid that maps the container's cells to the coordinates used to
+            define the bounding box.  May define a grid that is a super of the
+            container's cells.
+        bbox : `~lsst.geom.Box2I`
+            Bounding box that returned cells must overlap.
+
+        Returns
+        -------
+        grid_container : `GridContainer`
+            GridContainer with just the cells that overlap the bounding box.
+        """
+        bbox = bbox.clippedTo(grid.bbox)
+        offset = grid.index(bbox.getBegin())
+        last = grid.index(bbox.getMax())
+        end = Index2D(last.x + 1, last.y + 1)
+        shape = Index2D(end.x - offset.x, end.y - offset.y)
+        gc = GridContainer(shape, offset)
+        for index in gc.indices():
+            gc[index] = self[index]
+        return gc
+
+    def rebuild_transformed(self, transform: Callable[[T], T]) -> GridContainer:
+        """Return a GridContainer with the same shape and offset.
+
+        The cell values are created by applying a callback function to each
+        cell value in this object.
+
+        Parameters
+        ----------
+        transform: Callable[[T], T]
+            A callable function that takes a cell value and returns a new
+        """
+        gc = GridContainer(self.shape, self.offset)
+        for key in self.keys():
+            gc[key] = transform(self[key])
+        return gc
```

Comment:
We discussed to remove this method and just use the code wherever it was needed, but I found it was used in two places and it seemed easier to keep it than remove it. One input I'm looking for is whether we should return a new `GridContainer` instance or make these methods modify the given instance. Same question for `subset_overlapping`. Note that `GridContainer` is still something that's used only by this package, and I don't foresee any other package (say `descwl-coadd-task` or something similar) necessarily needing this class. They will be dealing with `SingleCellCoadd` and `MultipleCellCoadd` classes instead.